src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -37,28 +37,38 @@ class AppTasksResource @Inject() (service: MarathonSchedulerService,
 
   @GET
   @Timed
-  def indexJson(@PathParam("appId") appId: String,
+  def indexJson(@PathParam("appId") id: String,
                 @Context req: HttpServletRequest, @Context resp: HttpServletResponse): Response = {
-    doIfAuthorized(req, resp, ViewAppOrGroup, appId.toRootPath) { implicit principal =>
+    doIfAuthenticated(req) { implicit identity =>
       val taskMap = taskTracker.tasksByAppSync
 
-      def tasks(appIds: Set[PathId]): Set[EnrichedTask] = for {
+      def runningTasks(appIds: Set[PathId]): Set[EnrichedTask] = for {
+        runningApps <- appIds.filter(taskMap.hasAppTasks)
         id <- appIds
         health = result(healthCheckManager.statuses(id))
         task <- taskMap.marathonAppTasks(id)
       } yield EnrichedTask(id, task, health.getOrElse(Task.Id(task.getId), Nil))
 
-      val matchingApps = appId match {
+      id match {
         case GroupTasks(gid) =>
-          result(groupManager.group(gid.toRootPath))
-            .map(_.transitiveApps.map(_.id))
-            .getOrElse(Set.empty)
-        case _ => Set(appId.toRootPath)
+          val groupPath = gid.toRootPath
+          val maybeGroup = result(groupManager.group(groupPath))
+          doIfAuthorized(req, ViewGroup, maybeGroup, unknownGroup(groupPath)) { identity =>
+            val tasks = runningTasks(maybeGroup.get.transitiveApps.map(_.id))
```

Comment:
Even if get is safe here, we should never call it directly. Please see my comment.